crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1347,13 +1347,26 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
             SuiError::from("Unexpected new epoch number")
         );
 
-        let signed_epoch = SignedEpoch::new(new_committee, authority, secret, next_checkpoint);
+        let signed_epoch = SignedEpoch::new(
+            new_committee,
+            authority,
+            secret,
+            next_checkpoint,
+            latest_epoch.epoch_info(),
+        );
         self.tables
             .epochs
             .insert(&cur_epoch, &AuthenticatedEpoch::Signed(signed_epoch))?;
         Ok(())
     }
 
+    pub fn store_epoch_cert(&self, cert: CertifiedEpoch) -> SuiResult {
```

Comment:
Question: Is this always atomic, or should we expose an interface by which we add the update to a larger batch, that commits this and a bunch of other things? (if we never check for it by key - ie epoch num - I think we are ok).

Reply:
This should be atomic